Stats & Reader: followers should direct to site-stream

samouri opened the pull request:
I recently noticed two places with references to Reader or Reader-like things that don't actually link to Reader.

The followers stats module located at: `/stats/insights/`
<img width="349" alt="screen shot 2017-10-18 at 5 25 41 pm" src="https://user-images.githubusercontent.com/4656974/31743572-95b94cca-b429-11e7-998f-b14857c90ebe.png">


And also the followers list located at: `/people/followers/`
<img width="719" alt="screen shot 2017-10-18 at 5 27 38 pm" src="https://user-images.githubusercontent.com/4656974/31743623-bbbffb94-b429-11e7-8c84-a29c26623af9.png">

Stats names chosen:
1. `calypso_reader_stats_insights_followers_link_click`
2. `calypso_reader_people_followers_link_click`

Thanks @aaronyan for the help in naming!

todo:
- [x] to reader from insights
- [x] insights tracks
- [x] to reader from people
- [x] people tracks

samouri commented:
To Test
-------

- [ ] verify ux of both /people and the stats insights page
- [ ] verify both tracks events fire

fraying commented:
This is great, @samouri.

Bonus points (or maybe a separate PR) for adding follow links to the Followers page, maybe like so: 

![follow-followers](https://user-images.githubusercontent.com/24388/31783225-65276dba-b4b2-11e7-8ba3-3d0a9b2e3ff3.png)

(Pinging @jancavan for design review.)

It's always seemed weird to me that the list of followers on the Insights page has follow links, but the same info on the Followers page does not.

jancavan commented:
Nice work @samouri! 

> Bonus points (or maybe a separate PR) for adding follow links to the Followers page, maybe like so:

Good call. I wanted to see how it'd look if we left-aligned the Follow links like in Reader:
![screenshot 2017-10-19 11 17 22](https://user-images.githubusercontent.com/4924246/31786978-1f495a12-b4bf-11e7-9692-9836d6c4ed8e.png)

But it looks weird. I think because the "Remove" link is always present (as opposed to Settings which only shows up after Following):

![screenshot 2017-10-19 11 24 13](https://user-images.githubusercontent.com/4924246/31787295-22ad8ac4-b4c0-11e7-87b7-30ef8e237819.png)

jancavan commented:
I can confirm that the Author names now link to Reader and the events are being fired.

@samouri Did you happen to change this on the Stats Comment module as well? Or are Comments and Followers using the same component? It got updated in Comments too, which is good.

samouri commented:
> Or are Comments and Followers using the same component? It got updated in Comments too, which is good.

Turns out the same component is reused :O.  I had no idea!

  I'll rename the tracks event

samouri commented:
renamed the track event to `calypso_reader_stats_module_site_stream_link_click`

jancavan commented:
@samouri @aaronyan Should two different events be fired so we can determine which click came from which module? Or is there a way to segment that?

samouri commented:
@jancavan: decided to make it the same track but with a differentiating `moduleName` prop

blowery commented on client/my-sites/people/people-profile/index.jsx:
```diff
@@ -80,6 +81,8 @@ const PeopleProfile = React.createClass( {
 		return 'role-' + role;
 	},
 
+	handleLinkToReaderSiteStream: () => recordTrack( 'calypso_reader_people_followers_link_click' ),
```
hrm. this isn't really a reader event. probably `calypso_sites_people_followers_link_click` instead?

samouri replied:
sure

blowery commented on client/my-sites/stats/stats-list/stats-list-item.jsx:
```diff
@@ -216,8 +230,24 @@ const StatsListItem = React.createClass( {
+				}
 				itemLabel = (
-					<a onClick={ this.preventDefaultOnClick } href={ data.link }>
+					<a onClick={ onClickHandler } href={ href }>
```
isn't `href` out of scope here?

blowery commented on client/my-sites/stats/stats-list/stats-list-item.jsx:
```diff
@@ -216,8 +230,24 @@ const StatsListItem = React.createClass( {
+				if ( this.isFollowersModule ) {
+					const siteId = this.getSiteIdForFollow();
+					if ( siteId ) {
+						href = '';
```
why do we erase the href?

samouri replied:
i don't think we need to since i've added the `preventDefault()`

blowery replied:
over in reader, we only prevent when it's a default (left) mouse click with no modifiers held. That way, a user can option-click or right click on the link and open the original URL in a new tab or window.

samouri replied:
ah, ok ill update this

samouri commented:
couple changes:
1. added a new prop to the tracks events that indicates if it was opened in a new tab or not
2. fixed a problem where we were giving html tags more properties than they accept
3. fixed an issue where i was nesting anchor tags

blowery commented:
testing

blowery commented:
tested out fine, LGTM

blowery commented on client/my-sites/people/people-profile/index.jsx:
```diff
@@ -77,6 +79,23 @@ class PeopleProfile extends React.PureComponent {
+			return;
+		}
+
+		const blogId = get( this.props.user, 'follow_data.params.blog_id', false );
```
should probably bail if we get `false` here

samouri replied:
i already have the bail in calling code, but makes sense to have this fully independent

samouri replied:
bail added! ready for final 👀
